redis/connection.py

```diff
@@ -815,6 +815,8 @@ def can_read(self, timeout=0):
 
     def read_response(self, disable_decoding=False):
         """Read the response from a previously sent command"""
+        if not self._sock:
+            self.connect()
```

Comment:
Shouldn't this be added to the pub/sub code instead of adding this check for every redis request?

Reply:
Possibly.  read_response() is after all used in other places than the pub-sub.  I'll see if I can do this differently.

Reply:
Thanks for the suggestion.  Now the can_read/read_response interaction, along with an extra `connect()`, is wrapped in a lambda passed to `PubSub._execute()`